src/Orleans/Utils/Utils.cs

```diff
@@ -371,5 +372,139 @@ public static string GetStackTrace(int skipFrames = 0)
             return new System.Diagnostics.StackTrace(skipFrames).ToString();
 #endif
         }
+
+        private class ArraySlice<T> : IReadOnlyList<T>
```

Comment:
This already exists in the framework as ArraySegment. Please use that one instead.